packages/@dcl/ecs/src/engine/index.ts

```diff
@@ -59,10 +61,23 @@ function preEngine() {
     return entityContainer.removeEntity(entity)
   }
 
-  function defineComponent<T extends ISchema>(
-    componentId: number,
-    spec: T
+  function ensureComponentId(componentId?: number) {
+    if (!componentId) {
+      const CUSTOM_COMPONENTS_START_AT = 10e3
+      let i = CUSTOM_COMPONENTS_START_AT
+      while (componentsDefinition.get(i)) {
+        i++
+      }
+      return i
+    }
+    return componentId
+  }
+
+  function defineComponentFromSchema<T extends ISchema>(
+    spec: T,
+    componentIdDefined?: number
```

Comment:
im not sure about this. The component ID is an important thing to be created randomly.

I prefer to be a mandatory field and expose a method that is getAvailableComponentID or something like that, so the creator knows how this works and whats the pro/cons of doing this